### PR TITLE
Free blk list in kvset split

### PR DESCRIPTION
**Pull request by gsramdasi**
Signed-off-by: Gaurav Ramdasi <10132364+gsramdasi@users.noreply.github.com>

This addresses one source of the leaks that the sanitizer detected. The other is fixed as part of PR #437

## Issue(s) Addressed
NFSE-5346

### CODE REVIEW

File: lib/cn/kvset_split.c [hunk 1]
```diff
@@ -341,50 +341,48 @@ kblocks_split(
...
         }
 
         /* Add the source kblock to the purge list to be destroyed later */
-        err = blk_list_append(result->blks_purge, kblk->kb_kblk.bk_blkid);
-        if (err)
-            return err;
+        if (!err)
+            err = blk_list_append(result->blks_purge, kblk->kb_kblk.bk_blkid);
+
+        for (int i = LEFT; i <= RIGHT; i++)
```
Comment (tristan957) [inline lib/cn/kvset_split.c:377]: Might be more clear as two separate statements

**Review by tristan957 — APPROVED** [review]

File: lib/cn/kvset_split.c [hunk 2]
```diff
@@ -341,50 +341,48 @@ kblocks_split(
 
         /* split kblock at split_idx */
         err = kblock_split(&kbd, split_kobj, kblks, hlogs, vused);
-        if (err)
-            return err;
-        assert(kblks[LEFT].n_blks > 0 && kblks[RIGHT].n_blks > 0);
+
+        assert(err || (kblks[LEFT].n_blks > 0 && kblks[RIGHT].n_blks > 0));
```
Comment (nabeelmmd) [inline lib/cn/kvset_split.c:345]: nit: swap the operands so that the most common case comes first.

File: lib/cn/kvset_split.c [hunk 3]
```diff
@@ -341,50 +341,48 @@ kblocks_split(
...
 
         split_idx++;
     }
 
     /* Add kblocks in [split_idx, nkblks - 1] to the right kvset
      */
-    for (uint32_t i = split_idx; i < ks->ks_st.kst_kblks; i++) {
+    for (uint32_t i = split_idx; i < ks->ks_st.kst_kblks && !err; i++) {
         err = blk_list_append(&blks_right->kblks, ks->ks_kblks[i].kb_kblk.bk_blkid);
         if (err)
```
Comment (nabeelmmd) [inline lib/cn/kvset_split.c:387]: Just following this pattern, we could put line 390 and 391 under if (!err) and get rid of the return statement at  line 388.

**Review by nabeelmmd — APPROVED** [review]

**Comment by nabeelmmd** [comment]
Thanks for fixing this, Gaurav!

File: lib/cn/kvset_split.c (hunk 1 above)
Comment (gsramdasi) [reply lib/cn/kvset_split.c:377]: Yeah, I did it this way so it's consistent with the `blk_list_init()` calls above.

File: lib/cn/kvset_split.c (hunk 2 above)
Comment (gsramdasi) [reply lib/cn/kvset_split.c:345]: Yes. Done.

File: lib/cn/kvset_split.c (hunk 1 above)
Comment (tristan957) [reply lib/cn/kvset_split.c:377]: Could always change those, but it doesn't matter! Merge at will

**Review by alexttx — APPROVED** [review]

**Pull request merged**